diskqueue.go

```diff
@@ -693,6 +944,20 @@ func (d *diskQueue) handleReadError() {
 			d.name, badFn, badRenameFn)
 	}
 
+	if d.enableDiskLimitation {
+		if d.readFileNum == d.writeFileNum {
+			// we moved on to the next writeFile
+			d.totalDiskSpaceUsed = 0
+			d.writeMessages = 0
```

Comment:
It only renames the filename of bad file, do we need to update `d. totalDiskSpaceUsed` and `d. writeMessages`?

Reply:
Yup, I updated the code to improve readability, but if readFileNum == writeFileNum, then DiskQueue will skip to the next file. In this case, we need to set totalDiskSpaceUsed and writeMessages to 0, since everything was reset.